### PR TITLE
Fix/expansion panel list icon

### PR DESCRIPTION
**Pull request by funyin**
- Added parameter to change expansion icon of expansion panel
- Added parameter to show or hide expansion panel icon

### Issues
- Fixed compulsory icon that comes with ExpansionPaneList

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Renzo-Olivares** [comment]
Thanks for your contribution. This seems like a reasonable change, but we will need a test before we can land any change.

File: packages/flutter/lib/src/material/expansion_panel.dart [hunk 1]
```diff
@@ -352,16 +361,19 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
         context,
         _isChildExpanded(index),
       );
-
+        
+      var onTapIcon = !child.canTapOnHeader
```
Comment (Gabanelli) [inline packages/flutter/lib/src/material/expansion_panel.dart:365]: `onTapIcon` can be final

File: packages/flutter/lib/src/material/expansion_panel.dart [hunk 2]
```diff
@@ -352,16 +361,19 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
...
+              ? (bool isExpanded) => _handlePressed(isExpanded, index)
+              : null
```
Comment (Gabanelli) [inline packages/flutter/lib/src/material/expansion_panel.dart:367]: I think you missed a `;`

File: packages/flutter/lib/src/material/expansion_panel.dart [hunk 3]
```diff
@@ -352,16 +361,19 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
...
       Widget expandIconContainer = Container(
         margin: const EdgeInsetsDirectional.only(end: 8.0),
-        child: ExpandIcon(
+        child: icon==null?ExpandIcon(
```
Comment (Gabanelli) [inline packages/flutter/lib/src/material/expansion_panel.dart:370]: Format file to fix code spacing

**Review by Gabanelli — COMMENTED** [review]
I commented some suggestions to your PR

**Review by Renzo-Olivares — CHANGES_REQUESTED** [review]
Just a reminder that this PR still needs a test to be merged.

**Comment by Hixie** [comment]
@funyin Thanks for your contribution. I'm going to close this PR since it doesn't have a test and so can't land in its current state, but please don't hesitate to submit it again when you are ready! If you need help writing tests please don't hesitate to reach out to us on the Discord (link in the contributing guidelines, link to that is below), especially the #hackers-new channel.

**Pull request closed without merging**